.circleci/orbs/code-infra.yml

```diff
@@ -0,0 +1,173 @@
+version: 2.1
+
+commands:
+  install-deps:
+    description: 'Installs JavaScript dependencies using pnpm'
+    parameters:
+      browsers:
+        type: boolean
+        default: false
+        description: 'Set to true if you intend to install any browser (for example with playwright).'
+      package-overrides:
+        description: 'A space separated list of package@version to override during installation'
+        type: string
+        default: ''
+      ignore-workspace:
+        description: 'Set to true to ignore the pnpm workspace (useful for single-package repositories)'
+        type: boolean
+        default: false
+    steps:
+      - run:
+          name: Set npm registry public signing keys
+          command: |
+            echo "export COREPACK_INTEGRITY_KEYS='$(curl https://registry.npmjs.org/-/npm/v1/keys | jq -c '{npm: .keys}')'" >> $BASH_ENV
+      - when:
+          condition: << parameters.browsers >>
+          steps:
+            - run:
+                name: Install pnpm package manager
+                command: corepack enable
+      - when:
+          condition:
+            not: << parameters.browsers >>
+          steps:
+            - run:
+                name: Install pnpm package manager
+                # See https://stackoverflow.com/a/73411601
+                command: corepack enable --install-directory ~/bin
+      - run:
+          name: View install environment
```

Comment:
Wondering whether these three steps not make more sense in `mui-node`? Or maybe we need a `mui-node-browser` as well? 🤔

Reply:
But we use two different images right? Would that mean we'd have two different executors?

And should the `browser` parameter not just follow whichever executor is used? Would it make sense/be possible to create two executors, one for cimg/node and one for the playwright image. Then have a env var in both telling downstream jobs what they are (e.g. `MUI_EXECUTOR=node|browser`), then use this env var instead of passing the `browser` parameter to those jobs?